cmd/gitops/remove/run/cmd.go

```diff
@@ -0,0 +1,172 @@
+package run
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/weave-gitops/cmd/gitops/cmderrors"
+	"github.com/weaveworks/weave-gitops/cmd/gitops/config"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"github.com/weaveworks/weave-gitops/pkg/run"
+	"github.com/weaveworks/weave-gitops/pkg/run/session"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	"os"
+)
+
+type RunCommandFlags struct {
+	AllSessions bool
+
+	// Global flags.
+	Namespace  string
+	KubeConfig string
+
+	// Flags, created by genericclioptions.
+	Context string
+}
+
+var flags RunCommandFlags
+
+var kubeConfigArgs *genericclioptions.ConfigFlags
+
+func RunCommand(opts *config.Options) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Remove GitOps Run sessions",
+		Long:  "Remove GitOps Run sessions",
+		Example: `
+# Remove the GitOps Run session "dev-1234" from the "flux-system" namespace
+gitops remove run --namespace flux-system dev-1234
+
+# Remove all GitOps Run sessions from the default namespace
+gitops remove run --all-sessions
+
+# Remove all GitOps Run sessions from the dev namespace
+gitops remove run -n dev --all-sessions
+`,
+		PreRunE: removeRunPreRunE(opts),
+		RunE:    removeRunRunE(opts),
+
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+		DisableAutoGenTag: true,
+	}
+
+	cmdFlags := cmd.Flags()
+
+	cmdFlags.BoolVar(&flags.AllSessions, "all-sessions", false, "Remove all GitOps Run sessions")
+
+	kubeConfigArgs = run.GetKubeConfigArgs()
+
+	kubeConfigArgs.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func getKubeClient(cmd *cobra.Command, args []string) (*kube.KubeHTTP, *rest.Config, error) {
```

Comment:
💡  Is there any difference between this function and the `getKubeClient` function here?
https://github.com/weaveworks/weave-gitops/blob/session-remove/cmd/gitops/beta/run/cmd.go#L181

I only see the following log message removed `log.Actionf("Checking for a cluster in the kube config ...")` in the remove cmd.go version comparing to the run cmd.go version. Can we just remove this log message and combine these two functions?

Reply:
Yes, I need to refactor it somehow. Haven't got a good idea how should I do it yet.